dispatch/base.h

```diff
@@ -135,6 +135,12 @@
 #else
 #define DISPATCH_EXPORT extern __declspec(dllexport)
 #endif
+#elif defined(__DISPATCH_BUILDING_SWIFT_DISPATCH__)
+#if defined(__cplusplus)
+#define DISPATCH_EXPORT extern "C"
+#else
+#define DISPATCH_EXPORT extern
+#endif
```

Comment:
This is wrong for DLL builds.

Reply:
This is not enabled when we build the DLL API. This is only enabled while we are building the swiftDispatch.o. Previously the problem was that we compiled swiftDispatch.o with the Swift compiler and the Swift compiler treated all references to libdispatch C API functions as references that should be imported from a DLL. But that's not what we want because we statically link swiftDispatch.o and the other libdispatch components together into a single DLL. So the linker ended up complaining that we are trying to import local definitions.

We now define DISPATCH_EXPORT as extern while we are building swiftDispatch.o to ensure that the Swift compiler will treat references to libdispatch as local references which is what we want because swiftDispatch.o and all the rest of libdispatch is statically linked together into a single DLL.

Code which uses this DLL will not see the above definition of DISPATCH_EXTERN.